exporters/etw/include/opentelemetry/exporters/etw/etw_tracer.h

```diff
@@ -433,12 +410,42 @@ class Tracer : public opentelemetry::trace::Tracer,
     opentelemetry::trace::SpanContext parentContext = GetCurrentSpan()->GetContext();
     if (nostd::holds_alternative<opentelemetry::trace::SpanContext>(options.parent))
     {
-      auto span_context = nostd::get<opentelemetry::trace::SpanContext>(options.parent);
-      if (span_context.IsValid())
+      auto spanContext = nostd::get<opentelemetry::trace::SpanContext>(options.parent);
+      if (spanContext.IsValid())
       {
-        parentContext = span_context;
+        parentContext = spanContext;
       }
     }
+    auto traceId = parentContext.IsValid() ? parentContext.trace_id() : traceId_;
+
+    // Sampling based on attributes is not supported for now, so passing empty below.
+    std::map<std::string, int> emptyAttributes = {{}};
+    opentelemetry::sdk::trace::SamplingResult sampling_result =
+        GetSampler(tracerProvider_)
+            .ShouldSample(parentContext, traceId, name, options.kind,
+                          opentelemetry::common::KeyValueIterableView<std::map<std::string, int>>(
+                              emptyAttributes),
+                          links);
+
+    opentelemetry::trace::TraceFlags traceFlags =
+        sampling_result.decision == opentelemetry::sdk::trace::Decision::DROP
+            ? opentelemetry::trace::TraceFlags{}
+            : opentelemetry::trace::TraceFlags{opentelemetry::trace::TraceFlags::kIsSampled};
+
+    auto spanContext =
+        std::unique_ptr<opentelemetry::trace::SpanContext>(new opentelemetry::trace::SpanContext(
+            traceId, GetIdGenerator(tracerProvider_).GenerateSpanId(), traceFlags, false,
+            sampling_result.trace_state
+                ? sampling_result.trace_state
+                : parentContext.IsValid() ? parentContext.trace_state()
+                                          : opentelemetry::trace::TraceState::GetDefault()));
+
+    if (sampling_result.decision == sdk::trace::Decision::DROP)
+    {
+      auto noopSpan = nostd::shared_ptr<trace::Span>{
```

Comment:
Is it need to create the noopSpan with the new `spanContext`?

Reply:
yes, it is required. While noopSpan is not recorded/sampled, it would be used as a parent to decide whether the child of this noopSpan should be recorded or not. And for that to happen, the parent should have a valid span context.

Reply:
So int he drop case, only the `traceFlags` matter in the `spanContext`? If so, could we create a static `spanContext` for that? Just want to see if the `spanContext` allocated could be avoided.

Reply:
We don't know what part of `spancontext` can be used in the sampling decisions, as the application can come with its own custom sampler, which can possibly use trace_id from parent context to decide what to do with child span.

Reply:
Also the sdk [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.0.0/specification/trace/sdk.md#sdk-span-creation) states this

``
Generate a new span ID for the Span, independently of the sampling decision. This is done so other components (such as logs or exception handling) can rely on a unique span ID, even if the Span is a non-recording instance.
``

Reply:
Thanks for the details which makes sense.